File: app/src/main/java/com/ikvych/cocktail/service/DrinkDao.java
package com.ikvych.cocktail.service;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.ikvych.cocktail.model.Drink;

import java.util.List;

@Dao
public interface DrinkDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Drink drink);

    @Update
    void update(Drink drink);

    @Delete
    void delete(Drink drink);

    @Query("SELECT * FROM drink")
    LiveData<List<Drink>> getAllDrink();
}

File: app/src/main/java/com/ikvych/cocktail/data/db/impl/dao/DrinkDao.kt
package com.ikvych.cocktail.data.db.impl.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.ikvych.cocktail.data.db.impl.dao.base.BaseDao
import com.ikvych.cocktail.data.db.model.Drink

@Dao
interface DrinkDao : BaseDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(drink: Drink)

    @Query("SELECT * FROM drink WHERE drink_of_day=:stringDate")
    fun findDrinkOfTheDay(stringDate: String) : Drink?

    @Query("SELECT * FROM drink WHERE id_drink=:drinkId")
    fun findDrinkLiveDataById(drinkId: Long) : LiveData<Drink?>

    @Query("SELECT * FROM drink WHERE id_drink=:drinkId")
    suspend fun findDrinkById(drinkId: Long) : Drink

    @Query("SELECT * FROM drink WHERE str_drink=:drinkName")
    fun findDrinkByName(drinkName: String) : Drink

    @Query("SELECT * FROM drink")
    fun getAllDrinks() : List<Drink>

    @Query("SELECT * FROM drink ORDER BY created DESC")
    fun getAllDrinksLiveData() : LiveData<List<Drink>>

    @Query("SELECT * FROM drink WHERE is_favorite = 0 ORDER BY created DESC")
    fun getAllFavoriteDrinksLiveData() : LiveData<List<Drink>>

}